src/consumers/consumers.c

```diff
@@ -2,15 +2,118 @@
 #include <string.h>
 #include <chan/chan.h>
 
-#include "../statsd-parsers.h"
+#include "../../statsd-parsers/statsd-parsers.h"
 #include "../../utils/utils.h"
+#include "consumers.h"
+#include "./metric-types/counter/counter.h"
+#include "./metric-types/duration/duration.h"
+#include "./metric-types/gauge/gauge.h"
 
+/**
+ * Flag to capture USR1 signal event, which is supposed mark request to output currently recorded data in debug file 
+ */
+static int g_output_requested = 0;
```

Comment:
Oh now I understand why you use USR1. Disregard my previous comments about `killall` then.

Reply:
Now here is a problem - g_output_requested variable is not thread safe, it will be set from a signal by a signal handler thread and then accessed by the consumer thread.

I suggest to use a design pattern here (can't remember the name), the thing is to use channels and special data structure which can be utilized for special "message" with a meaning. In this case "dump debug output" but you could utilize those "messages" for different meanings. You could introduce new "type" for that, something that is not used by statsd or any known extension.

Or you can ensure the variable is thread safe - also an option :-)

Reply:
I could make it so, that for example, it would be possible to request such debug output via datagram of its own type - "output:1|d" - such parsed datagram would trigger the consumer to output N = 0b001 hash tables to file where N is bitfield of hashtables (001 = only counter, 010 = only gauge, 100 = only duration), but isn't that overly complicated?

Reply:
Variable is behind lock by 893df1f

Reply:
You make your choice, lock is probably better yes.